doc/user/unit-testing.md

```diff
@@ -1,132 +1,287 @@
+    cl := fake.NewFakeClient(objs...)
+  
+    // Create a ReconcileMemcached object with the scheme and fake client.
+    r := &ReconcileMemcached{client: cl, scheme: s}
+    
+    // Mock request to simulate Reconcile() being called on an event for a
+    // watched resource .
+    req := reconcile.Request{
+        NamespacedName: types.NamespacedName{
+            Name:      name,
+            Namespace: namespace,
+        },
+    }
+ 
+    res, err := r.Reconcile(req)
+    if err != nil {
+       t.Fatalf("reconcile: (%v)", err)
+    }
+    
+    // Check the result of reconciliation to make sure it has the desired state.
+    if !res.Requeue {
+        t.Error("reconcile did not requeue request as expected")
+    }
+    
+    // Check if deployment has been created and has the correct size.
+    dep := &appsv1.Deployment{}
+    err = r.cl.Get(context.TODO(), req.NamespacedName, dep)
+    if err != nil {
+        t.Fatalf("get deployment: (%v)", err)
+    }
+    
+    // Check if the quantity of Replicas for this deployment is equals the specification
+    dsize := *dep.Spec.Replicas
+    if dsize != replicas {
+        t.Errorf("dep size (%d) is not the expected size (%d)", dsize, replicas)
+    }
 }
 ```
 
-The above tests if:
-- `Reconcile()` fails to find a Deployment object.
-- A Deployment is created.
-- The request is requeued in the expected manner.
+**The above tests check if:**
+
+- `Reconcile()` fails to find a Deployment object
+- A Deployment is created
+- The request is requeued in the expected manner
 - The number of replicas in the created Deployment's spec is as expected.
 
-A unit test checking more cases can be found in our [`samples repo`][code_test_example].
+**NOTE**: A unit test checking more cases can be found in our [`samples repo`][code_test_example].
 
-[doc_e2e_test]:https://github.com/operator-framework/operator-sdk/blob/2f772d1dc2340dd19bdc3ec8c2dc9f0f77cc8297/doc/test-framework/writing-e2e-tests.md
-[doc_client]:https://github.com/operator-framework/operator-sdk/blob/5c50126e7a112d67826894997eca143e12dc165f/doc/user/client.md
-[doc_cr_fake_client]:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/fake
-[repo_memcached_reconcile]:https://github.com/operator-framework/operator-sdk-samples/blob/4c6934448684a6953ece4d3d9f3f77494b1c125e/memcached-operator/pkg/controller/memcached/memcached_controller.go#L82
-[doc_reconcile]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler
-[code_test_example]: https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/pkg/controller/memcached/memcached_controller_test.go#L25
+## Testing with 3rd Party Resources
+
+You may have added third-party resources in your operator as described in the [`Advanced Topics section of the user guide`][user_guide]. In order to create a unit-test to test this kind of resources, it is required to add the Schema as it was done for your operator project.
```

Comment:
@camilamacedo86 Just to be clear, you only need to pass in the updated scheme to your Reconciler if your handler is actually using it.
If say I don't have a `scheme *runtime.Scheme` in my Reconciler struct then I don't need to do this.
https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/pkg/controller/memcached/memcached_controller.go#L74

Also:
```suggestion
You may have added third-party resources in your operator as described in the [`Advanced Topics section of the user guide`][user_guide]. In order to create a unit-test to test these kinds of resources, it might be necessary to update the Scheme with the third-party resources and pass it to your Reconciler.
```

Reply:
👍